download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/cache/Index.java

```diff
@@ -0,0 +1,32 @@
+package com.googlecode.download.maven.plugin.internal.cache;
+
+/**
+ * Cache index.
+ * @author Paul Polishchuk
+ * @since 1.3.1
+ */
+interface Index {
+
+    /**
+     * Adds given path to the index using url parameter as a key.
+     * @param url index key
+     * @param path index value
+     */
+    void put(String url, String path);
```

Comment:
You say you don't like wrapper, but wrap a map into an Index ;) Typing is quite often about wrapping a lower-level structure and rewording it's operations (and providing new ones). The question isn't really what the new types does more than the original one, but whether readers/implementors/parser/static analyzers.. will more easily understand what's expected.
Handling exception on URI is actually an almost free way to put some validation on the argument passed by user, and if failing, will fail to user with a more explicit error message. Maven actually supports natively the parameter type java.net.URL https://maven.apache.org/guides/plugin/guide-java-plugin-development.html#URLs , so it means that there is already some checks done by Maven while reading plugin configuration.